magento/magento2
# Fix the issue with "Shipping address is not set" exception

The exception is being thrown instead of redirecting the customer to New Shipping Address form.

### Description
The issues appeared because of the interrupted shipment assignment process.
As result:
1. The "Shipping address is not set" exception (``Magento\Framework\Exception\StateException``) is not caught and the customer is not redirected to ``multishipping/checkout_address/newShipping`` page.
2. The incorrectly assigned record appears in ``quote_address`` table due to the interrupted process of quote saving by ``Magento\Framework\Exception\StateException``. It causes the additional error on Shopping Cart page, after performing the steps mentioned in *Manual testing scenarios*. (Fixed in: 6f88af5ffe9d4bd633d70aa5ca84d2f15b6a52f5)

### Fixed Issues 
1. magento/magento2#16555: "Shipping address is not set" exception in Multishipping Checkout.
2. magento/magento2#16555: "SQLSTATE[23000]: Integrity constraint violation" when trying to access Shopping Cart.

### Manual testing scenarios
1. Log In as Customer.
2. Add Product to Cart.
3. Proceed to Checkout.
4. Fill Shipping Address data, click Next (do not place Order)
5. Return to the Storefront.
6. Go to the Shopping Cart.
7. Click Check Out with Multiple Addresses. (Error)
8. Go to the Shopping Cart page again. (Error)

### Contribution checklist
